test-data/unit/check-unreachable-code.test

```diff
@@ -500,6 +500,82 @@ reveal_type(h)  # N: Revealed type is "builtins.bool"
 [builtins fixtures/ops.pyi]
 [out]
 
+[case testConditionalValuesBinaryOps]
+# flags: --platform linux
+import sys
+
+t_and_t = (sys.platform == 'linux' and sys.platform == 'linux') and 's'
+t_or_t = (sys.platform == 'linux' or sys.platform == 'linux') and 's'
+t_and_f = (sys.platform == 'linux' and sys.platform == 'windows') and 's'
+t_or_f = (sys.platform == 'linux' or sys.platform == 'windows') and 's'
+f_and_t = (sys.platform == 'windows' and sys.platform == 'linux') and 's'
+f_or_t = (sys.platform == 'windows' or sys.platform == 'linux') and 's'
+f_and_f = (sys.platform == 'windows' and sys.platform == 'windows') and 's'
+f_or_f = (sys.platform == 'windows' or sys.platform == 'windows') and 's'
+reveal_type(t_and_t) # N: Revealed type is "Literal['s']"
```

Comment:
Hmm inferring `Literal['s']` as the type seems inconsistent and arguably a bug (not directly related to this PR). For example, `a = 'y'` causes the inferred type of `a` to be `str`, not a literal type. Can you update the test cases to not rely on this glitch, e.g by replacing `'s'` with `str()`?

Reply:
I just borrowed the idea from a preceding test to keep them aligned, but yes, let's fix both then.